src/EditDialog.cpp

```diff
@@ -21,6 +21,591 @@
 #include <QPainter>
 #include <QClipboard>
 
+//  The following section of code should really be in a separate class
+//  but this is only a test!
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <time.h>
+
+#define INCREMENT 4
+
+#define ERROR_NONE                   0
+#define ERROR_INSUFFICIENT_MEMORY    1
+#define ERROR_INVALID_HEADER         2
+#define ERROR_INVALID_TRAILER        3
+#define ERROR_INVALID_OFFSET         4
+#define ERROR_INVALID_OBJECT_LENGTH  5
+#define ERROR_INVALID_REFERENCE      6
+
+typedef struct KEY {
+  struct OBJECT *key;
+  struct OBJECT *value;
+} KEY;
+
+typedef struct OBJECT {
+  int type;
+  int length;
+  union {
+    time_t date;
+    double real;
+    long integer;
+    unsigned long uid;
+    unsigned char binary[1];
+    char text[1];
+    short int utf16[1];
+    int refs[1];
+    KEY keys[1];
+    struct OBJECT *objects[1];
+  } data;
+} OBJECT;
+
+typedef struct CONFIG {
+  int bytesPerOffset;
+  int bytesPerReference;
+  int objectCount;
+  int rootObjectReference;
+  int offsetTableOffset;
+  size_t bufferLength;
+  size_t outputBufferLength;
+  size_t outputBufferIn;
+  long *offsetTable;
+  unsigned char *buffer;
+  unsigned char *outputBuffer;
+} CONFIG;
+
+int createObject(int type, int length, int extra, OBJECT **ptr2);
+int readObject(CONFIG *cfg, long offset, OBJECT **ptr2);
+
+static int indent = 2;
+
+int outputText(CONFIG *cfg, const char *text)
+{
+  size_t textLength = strlen((const char *)text);
+  size_t availableSpace = cfg->outputBufferLength - cfg->outputBufferIn;
+  while (textLength >= availableSpace) {
+    cfg->outputBufferLength += textLength + 1024;
+    cfg->outputBuffer = (unsigned char *)realloc(cfg->outputBuffer, cfg->outputBufferLength);
+    if (cfg->outputBuffer == NULL)
```

Comment:
Is `nullptr` more modern? :)

Reply:
NULL is C, nullptr is C++, not sure if it is now also in C